packages/anu-vue/src/components/loader/ALoader.vue

```diff
@@ -0,0 +1,118 @@
+<script setup lang="ts">
+import { ATypography } from '../typography'
+import ASpinner from './ASpinner.vue'
+import { loaderProps } from './props'
+import { isTypographyUsed } from '@/components/typography/utils'
+import { ConfigurableValue, useConfigurable } from '@/composables/useConfigurable'
+import { useLayer } from '@/composables/useLayer'
+
+const props = defineProps(loaderProps)
+
+defineOptions({
+  name: 'ALoader',
+})
+
+const slots = useSlots()
+const attrs = useAttrs()
+
+const loading = toRef(props, 'loading')
+
+const { getLayerClasses } = useLayer()
+const { styles, classes } = getLayerClasses(
```

Comment:
Ok, I tried the `useColor` composable but I can't see how it could resolve the problem with the `light` variant.

The main difficulty is I want the `ALoader` following the color of the parent component. With `bg-$a-layer-color`, I got the hsla color of the parent component **with his opacity** and I can't override, it's not really the color prop. It's the same with `text-$a-layer-color`. So If I want to create an overlay which hide the content, it's not possible with `bg-$a-layer-color` because it contains opacity of 0.15 for the light variant for example.
Also, the `fill` variant got a background with a plain color and white text. But the `text` variant got a white background with text color. It's not possible with `bg-$a-layer-color` and `text-$a-layer-color` to obtain the real background color of the parent component neither the text color. (`bg-$a-layer-color` in the `text` variant return the color from the prop, not white)

JS can't resolve this problem, only CSS.

I found a workaround and created a mask that hide the content. `bg-$a-layer-color` works like that but it's a bit tricky. It's on this commit https://github.com/jd-solanki/anu/pull/119/commits/fa16205a2bd2caf3c8a784a14618629f5fa15415

All of this will be really much cleaner with the `:has()` selector. I tried that and it looks fine.
But this selector is not supported in firefox without a polyfill...

Reply:
I haven't finalized this issue yet. I am brainstorming and trying to find the best solution.

Reply:
Ok, no worry, I need to take a little break with anu for next days. I have plenty of other stuff to do...